omnipod/src/main/java/info/nightscout/androidaps/plugins/pump/omnipod/OmnipodPumpPlugin.java

```diff
@@ -202,82 +258,42 @@ protected void onStart() {
                             (event.isChanged(getResourceHelper(), R.string.key_omnipod_pod_debugging_options_enabled)) ||
                             (event.isChanged(getResourceHelper(), R.string.key_omnipod_beep_smb_enabled)) ||
                             (event.isChanged(getResourceHelper(), R.string.key_omnipod_timechange_enabled)))
-                        rileyLinkOmnipodService.verifyConfiguration();
+                        aapsOmnipodManager.reloadSettings();
                 }, fabricPrivacy::logException)
         );
-        disposable.add(rxBus
+        disposables.add(rxBus
                 .toObservable(EventAppInitialized.class)
                 .observeOn(Schedulers.io())
                 .subscribe(event -> {
                     // See if a bolus was active before the app previously exited
                     // If so, add it to history
                     // Needs to be done after EventAppInitialized because otherwise, TreatmentsPlugin.onStart() hasn't been called yet
                     // so it didn't initialize a TreatmentService yet, resulting in a NullPointerException
-                    if (sp.contains(OmnipodConst.Prefs.ActiveBolus)) {
-                        String activeBolusString = sp.getString(OmnipodConst.Prefs.ActiveBolus, "");
+                    if (sp.contains(OmnipodStorageKeys.Prefs.ActiveBolus)) {
+                        String activeBolusString = sp.getString(OmnipodStorageKeys.Prefs.ActiveBolus, "");
                         aapsLogger.warn(LTag.PUMP, "Found active bolus in SP: {}. Adding Treatment.", activeBolusString);
                         try {
-                            ActiveBolus activeBolus = omnipodUtil.getGsonInstance().fromJson(activeBolusString, ActiveBolus.class);
+                            ActiveBolus activeBolus = aapsOmnipodUtil.getGsonInstance().fromJson(activeBolusString, ActiveBolus.class);
                             aapsOmnipodManager.addBolusToHistory(activeBolus.toDetailedBolusInfo(aapsLogger));
                         } catch (Exception ex) {
                             aapsLogger.error(LTag.PUMP, "Failed to add active bolus to history", ex);
                         }
-                        sp.remove(OmnipodConst.Prefs.ActiveBolus);
+                        sp.remove(OmnipodStorageKeys.Prefs.ActiveBolus);
                     }
                 }, fabricPrivacy::logException)
         );
-
-        super.onStart();
```

Comment:
Yes, this was in a strange place before (not the first statement in the method) because the RileyLinkOmnipodService was bound to the context in PumpPluginAbstract, and some other code in onStart() depended on that service. Now that we don't use PumpPluginAbstract anymore, I moved it to be the first statement in the method, which is more appropriate.